contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,291 @@
+        _setupRole(MANAGER_ROLE, _vault);
+
+        vault = _vault;
+        ryskLqPool = IRyskLiquidityPool(_ryskLiquidityPool);
+        underlying = IERC20(_underlying);
+
+        underlying.approve(_ryskLiquidityPool, type(uint256).max);
+    }
+
+    //
+    // Modifiers
+    //
+
+    modifier onlyManager() {
+        if (!hasRole(MANAGER_ROLE, msg.sender))
+            revert StrategyCallerNotManager();
+        _;
+    }
+
+    //
+    // Ownable
+    //
+
+    /**
+     * Transfers ownership of the Strategy to another account,
+     * revoking previous owner's ADMIN role and setting up ADMIN role for the new owner.
+     *
+     * @notice Can only be called by the current owner.
+     *
+     * @param _newOwner The new owner of the contract.
+     */
+    function transferOwnership(address _newOwner)
+        public
+        override(Ownable)
+        onlyOwner
+    {
+        if (_newOwner == address(0x0)) revert StrategyOwnerCannotBe0Address();
+        if (_newOwner == msg.sender)
+            revert StrategyCannotTransferOwnershipToSelf();
+
+        _transferOwnership(_newOwner);
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newOwner);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 shares = _underlyingToShares(_amount);
+
+        _resetPendingWithdrawalIfEpochAdvanced();
+
+        // because the vault can call updateInvested => withdrawToVault multiple times during the same epoch,
+        // and rysk liquidity pool aggregates withdrawal amounts inititated during the same epoch,
+        // we cannot withdraw less than already pending amount
+        if (pendingWithdrawal.shares >= shares) return;
```

Comment:
How long does an epoch take? And can we cancel a withdrawal request?

Reply:
1. We cannot know how long an epoch will last since that depends on an external call to executeEpochCalculation function in the LiquidityPool contract. This function can only be called by "keepers", which are privileged users with the responsibility to move the epochs forward (and some more maintenance roles). No idea also what an approximate epoch duration could be.

2. There is no option to cancel a withdrawal request since really a withdrawal request does not exist. When we initiate a withdrawal, there is a withdrawal receipt "created" for the caller. A withdrawal receipt is used only to track how much funds a caller wishes to withdraw in the next epochs and there can be only 1 withdrawal receipt per caller. If the withdrawal isn't completed nothing will be affected, it "lives" indefinitely. If we initiate another withdrawal in a new epoch and without completing the previous one, the previous withdrawal receipt will be overridden, meaning the amount "marked" to be withdrawn on the receipt will be set to the new one.

Reply:
The end of what u said is no longer the case, if the withdrawal is queued that will now always be honoured